File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) 
    {
        // Most Optimized Solution using Xor..
        // Xor removes the duplicates..
        int sum1 = 0;
        int sum2 = 0;
        for(int i = 0; i <= nums.size(); i++){
            if(i != nums.size()) sum1 ^= nums[i];
            sum2 ^= i;
        }
        return sum2 ^ sum1;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        Map<Integer, Boolean> myMap = new HashMap<>();
        int n = nums.length;
        for(int i = 0; i < n; i++) {
            myMap.put(nums[i], true);
        }
        for(int i = 0; i <= n; i++) {
            if(!(myMap.containsKey(i))){
                return i;
            }
        }
        return -1;
    }
}